### ISSUE TITLE
Calculating evidence  with exp has the highest f1 value, contradicting the shortcomings of softmax

### ISSUE THREAD

**Issue by han-yuexing** [Leon-bit-9527/ENER#3]
When I was reproducing the code, I found that the highest value of f1 was obtained with exp when calculating the evidence e. Then, if exp is used, the value of b is very similar to that calculated with softmax, so isn't it contradictory to the shortcomings of softmax as mentioned in the paper from EDL 2018 (Softmax is also notorious with inflating the probability of the predicted class as a result of the exponent employed on the neural net outputs), could you please explain?

**Comment by zhzhengit** [comment]
Thank you very much for your interest in our work. In response to your question, we would like to clarify your question:
By default, we use the exp exponential function as the calculated evidence e, and we indeed achieve the optimal F1 value. However, the first difference is that softmax normalizes the probability distribution to the 0-1 interval, and the sum of belief masses in all categories is not equal to 1, which shows that if there is little evidence, the sum of belief masses in all categories is not equal to 1. The classification dimension may be much less than one, thus increasing the uncertainty mass fraction.
In addition, this method needs to be learned in conjunction with our optimization method (Eq. 7 in the paper) to achieve better results. If you want to compare the difference between Softmax and belief mass, it may be clearer under the settings of SpanNER+CE+Softmax and E-NER+belief mass.

Thank you again for your interest in our work and please contact me if you have any further questions.

**Comment by han-yuexing** [comment]
Thanks for your answer about the difference between Softmax and belief mass. Another question I have is why exp would work better than relu as an exponential function, isn't this a contradiction to the idea of the paper that proposed EDL in 2018 (Softmax is also notorious with inflating the probability of the predicted class as a result of the exponent employed on the neural net outputs)

**Comment by han-yuexing** [comment]
Besides,  the paper that proposed EDL in 2018 said"We have observed the losses in Equations 3 and 4 to generate excessively high belief masses for classes and exhibit relatively less stable performance than Equation 5."
Could you please explain why you chose formula 3 directly?

**Comment by zhzhengit** [comment]
> Thanks for your answer about the difference between Softmax and belief mass. Another question I have is why exp would work better than relu as an exponential function, isn't this a contradiction to the idea of the paper that proposed EDL in 2018 (Softmax is also notorious with inflating the probability of the predicted class as a result of the exponent employed on the neural net outputs)

We have considered your question. Sensoy's research [1] showed that logits can be converted into non-negative evidence using functions such as exp or softplus. According to this non-negative transformation rule, you can try this step with more functions.

[1] Sensoy M, Kaplan L, Kandemir M. Evidential deep learning to quantify classification uncertainty[J]. Advances in neural information processing systems, 2018, 31.

**Comment by han-yuexing** [comment]
> Besides, the paper that proposed EDL in 2018 said"We have observed the losses in Equations 3 and 4 to generate excessively high belief masses for classes and exhibit relatively less stable performance than Equation 5." Could you please explain why you chose formula 3 directly?

Thanks for the answer to the first question.My last question is why you chose formula 3 directly in Sensoy's research[1]?
